### ISSUE TITLE
handle length restriction is too short.

### ISSUE THREAD

**Issue by itaru2622** [bluesky-social/atproto#2391]
**Describe the bug**

PDS restricts the length of the handle to 30 at below, which is too short for practical purposes.
below line (handle.length) means total length of user name part + FQDN part  is up to 30.

[RFC1035](https://datatracker.ietf.org/doc/html/rfc1035#section-2.3.4) defines maximum length of FQDN is 255. then, it needs relax restriction on handle length, up to 255.

https://github.com/bluesky-social/atproto/blob/f5c830a33b6774c4572973df103d4a861bbc5e7d/packages/pds/src/handle/index.ts#L89-L90

**To Reproduce**

```bash
# send request
curl -L -X POST 'https://pds-subdomain.mydomain.com/xrpc/com.atproto.server.createAccount' \
-H 'Content-Type: application/json'  -H 'Accept: application/json' \
-d '{ "email": "user567890@mydomain.com" ,"handle": "user567890.pds-subdomain.mydomain.com", "password": "shortpass" }' 

# result:
 {"error":"InvalidHandle","message":"Handle too long"}
```

1. starts PDS and sends above createAccount request.

NOTE: this restriction is not raised when testing with atproto/packages/dev-env.

**Expected behavior**

the above request should succeed since FQDN part meets RFC condition.

**Details**

use PDS docker image distributed at https://github.com/bluesky-social/pds or https://github.com/bluesky-social/atproto/tree/main/services/pds
- Operating system:
- Node version:

**Additional context**

**Comment by bnewbold** [comment]
Just to be clear, the atproto spec itself is quite clear about handle syntax, and allows up to the full standard hostname length (253 characters): https://atproto.com/specs/handle

Valid handles are fully supported by the PDS for *external* domain registrations, and when interacting with accounts on other instances.

This specific limit in the PDS source code is for *locally registered* handles, for example on the `*.bsky.social`. We have this restriction because we think it results in more usable and human-friendly handles. Users who want a longer handle are welcome to use an external domain name.

I suspect the friction you might be encountering with this is when self-hosting a PDS with a longer base domain name, which eats in to the characters available for the overall handle. Probably the best step is to update the code to limit the length of the "registered" part of the handle (aka, the first segment of the handle). If we set that limit to 18 characters, the behavior will be the same for `*.bsky.handle`, but more appropriate for longer base domains.

**Comment by bnewbold** [comment]
@itaru2622 yup! exactly

**Comment by itaru2622** [comment]
This issue was fixed by #2392 and #2410 and merged into main branch, then closed.
